app/src/main/java/com/google/samples/apps/sunflower/compose/plantlist/PlantListScreen.kt

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.samples.apps.sunflower.compose.plantlist
+
+import androidx.compose.foundation.layout.PaddingValues
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyVerticalGrid
+import androidx.compose.foundation.lazy.grid.items
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.livedata.observeAsState
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import com.google.samples.apps.sunflower.viewmodels.PlantListViewModel
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.google.samples.apps.sunflower.R
+import com.google.samples.apps.sunflower.data.Plant
+
+@Composable
+fun PlantListScreen(
+    viewModel: PlantListViewModel = viewModel(),
+    onPlantClick: (Plant) -> Unit,
+) {
+    val plants by viewModel.plants.observeAsState(initial = emptyList())
+    PlantListScreen(plants = plants, onPlantClick = onPlantClick)
+}
+
+@Composable
+fun PlantListScreen(
+    plants: List<Plant>,
+    onPlantClick: (Plant) -> Unit = {},
+) {
+    LazyVerticalGrid(
+        columns = GridCells.Fixed(2),
+        modifier = Modifier.testTag("plant_list"),
+        contentPadding = PaddingValues(
+            horizontal = dimensionResource(id = R.dimen.card_side_margin),
+            vertical = dimensionResource(id = R.dimen.header_margin)
+        )
+    ) {
+        items(
+            items = plants,
+            key = { it.plantId }
+        ) { plant ->
+            PlantListItem(plant = plant) {
+                onPlantClick(plant)
```

Comment:
I think you defo could trim passing the entire `Plant` to plantId, if im not missing something

Reply:
I wrote it like this so that the type is more explicit/safe. The call site can also get more information other than the ID if it wants (though currently, that's not the case). I'll keep it as-is for now but if you feel strongly I can address in a separate PR.